mtv_dl.py

```diff
@@ -268,27 +270,43 @@ def serialize_for_json(obj: Any) -> str:
     else:
         raise TypeError('%r is not JSON serializable' % obj)
 
+if system == 'Windows':
+    @contextmanager
+    def flocked(fd, timeout=1.0, sleep=0.1):
+        yield
 
-@contextmanager
-def flocked(fd, timeout=1.0, sleep=0.1):
-    """ Contextmanager to lock a file descriptor for exclusive reading/writing. """
+    invalid_chars = '<>:"/\\|?*' + "".join(chr(i) for i in range(32))
 
-    remaining_time = timeout
-    try:
-        while True:
-            try:
-                fcntl.flock(fd, fcntl.LOCK_EX | fcntl.LOCK_NB)
-                yield
-            except IOError:
-                if remaining_time or not timeout:
-                    time.sleep(sleep)
+else:
+    import fcntl
+
+    @contextmanager
+    def flocked(fd, timeout=1.0, sleep=0.1):
+        """ Contextmanager to lock a file descriptor for exclusive reading/writing. """
+
+        remaining_time = timeout
+        try:
+            while True:
+                try:
+                    fcntl.flock(fd, fcntl.LOCK_EX | fcntl.LOCK_NB)
+                    yield
+                except IOError:
+                    if remaining_time or not timeout:
+                        time.sleep(sleep)
+                    else:
+                        raise
                 else:
-                    raise
-            else:
-                break
+                    break
 
-    finally:
-        fcntl.flock(fd, fcntl.LOCK_UN)
+        finally:
+            fcntl.flock(fd, fcntl.LOCK_UN)
+
+    invalid_chars = '/\0'
+
+invalidre = re.compile("[{}]".format(re.escape(invalid_chars)))
```

Comment:
To make clear this is a constant, please use uppercase and underscore naming.

I also see no good reason to use re.escape here: Better the compiled regex in the if-else above.